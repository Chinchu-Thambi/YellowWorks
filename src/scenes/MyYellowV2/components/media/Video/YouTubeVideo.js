/* eslint-disable react/prop-types */
import React from 'react';
import { Card } from '../../Containers';
import { DeleteIconButton, EditIconButton } from '../../IconButtons';

/**
 * @type {(videoID: string, onDelete: Function, onEdit: Function) => JSX.Element}
 */
const YouTubeVideo = ({ videoID, onDelete, onEdit }) => (
  <Card iconButtons={[<DeleteIconButton onClick={onDelete} />, <EditIconButton onClick={onEdit} />]}>
    <div className="overflow-hidden rounded-md h-full w-full">
      <iframe
        className="w-full h-full"
        src={`https://www.youtube.com/embed/${videoID}`}
        frameBorder="0"
        title="YouTube Video"
      />
    </div>
  </Card>
);

export default YouTubeVideo;
