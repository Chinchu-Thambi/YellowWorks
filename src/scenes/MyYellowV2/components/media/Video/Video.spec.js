/* globals describe, it, expect */

import React from 'react';
import { render } from 'enzyme';

import YouTubeVideo from './YouTubeVideo';

describe('YouTubeVideo', () => {
  it('YouTubeVideo', () => {
    const wrapper = render(<YouTubeVideo videoID="YCp2GZNkBSU" onEdit={() => 'edit'} onDelete={() => 'delete'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
