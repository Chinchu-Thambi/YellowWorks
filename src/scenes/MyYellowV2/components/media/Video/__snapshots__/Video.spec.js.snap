// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTubeVideo YouTubeVideo 1`] = `
<div
  class="bg-contrast-100 rounded-md shadow-sm text-contrast-500 w-full h-full flex flex-col sm:flex-row space-y-2.5 sm:space-y-0 sm:space-x-2.5 py-2 pl-2 pr-1 sm:py-4 sm:pl-4 sm:pr-2.5"
>
  <div
    class="flex-1"
  >
    <div
      class="overflow-hidden rounded-md h-full w-full"
    >
      <iframe
        class="w-full h-full"
        frameborder="0"
        src="https://www.youtube.com/embed/YCp2GZNkBSU"
        title="YouTube Video"
      />
    </div>
  </div>
  <div
    class="flex flex-row space-x-2 justify-end sm:flex-col sm:justify-start sm:space-x-0 sm:space-y-2"
  >
    <span>
      <button
        aria-label="Delete button"
        class="h-5 w-5 rounded-full shadow-md active:shadow-xs flex items-center justify-center cursor-pointer uppercase space-x-2 text-contrast-400 bg-white hover:bg-crimson-600 hover:text-crimson-100 active:bg-crimson-500 active:text-crimson-100"
        type="button"
      >
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash fa-w-14 "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </span>
    <span>
      <button
        aria-label="Edit button"
        class="h-5 w-5 rounded-full shadow-md active:shadow-xs flex items-center justify-center cursor-pointer uppercase space-x-2 text-contrast-400 bg-white hover:bg-contrast-500 hover:text-contrast-100 active:bg-contrast-400 active:text-contrast-100"
        type="button"
      >
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-pen fa-w-16 "
            data-icon="pen"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
    </span>
  </div>
</div>
`;
